### PR TITLE
Focus on defining programmatic API, referring to service docs for URL structure

### PR DESCRIPTION
**Pull request by djmitche**
See https://github.com/taskcluster/taskcluster/pull/2036 for where the URL structure docs went..

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,6 +18,30 @@ Requirements
...
 This is tested on and should run on any of Node.js `{8, 10}`.
 
+General Usage
+-------------
+
+While the capitalization and punctunation of the function names varies depending on the language, each language provides the following methods:
+
+| method | result |
+| --- | --- |
+| api(rootUrl, service, version, path) | `<rootUrl>/api/<service>/<version>/<path>` |
```
Comment (owlishDeveloper) [inline README.md]: I'd maybe use a hindley-milner notation? for clarity

#preferential

Comment (djmitche) [reply README.md]: *googles*

Comment (djmitche) [reply README.md]: It looks like that's a type system?  Can you give me an example of what you mean?

**Comment by petemoore** [comment]
Looks good, many thanks!

**Pull request merged**